### ISSUE TITLE
Redesign Context menu 🔥

### ISSUE THREAD

**Issue by ad1992** [excalidraw/excalidraw#5887]
The context menu is becoming bigger and overloaded with more and more options so we need to revamp it. 

Let's discuss the ideas here on how we can improve it

**Comment by adarkforce** [comment]
Hi, @ad1992 

An idea:

![image](https://user-images.githubusercontent.com/50418432/204129704-deb2771c-0bf9-4ddf-b946-1c85e0b5e9da.png)

Or the "copy/paste as..." actions could be all grouped together, as Figma does:

![Schermata 2022-11-27 alle 11 09 52](https://user-images.githubusercontent.com/50418432/204129761-a8fdede1-defd-4bf6-93ff-56894c6f731a.png)

**Comment by dwelle** [comment]
@adarkforce yes, this is what we'll want to do. Probably best to use https://www.radix-ui.com/docs/primitives/components/context-menu for this.

But, before we do that, this should land first: https://github.com/excalidraw/excalidraw/issues/5926 (in progress)

**Comment by dwelle** [comment]
Should be implemented using [@radix-ui/react-context-menu](https://www.radix-ui.com/docs/primitives/components/context-menu), or possibly reuse [@radix-ui/react-dropdown-menu](@radix-ui/react-dropdown-menu) (we'll soon be using for main menu) if it turns out that the contextmenu adds too much size on top of their dropdownmenu.

**Comment by jamiecjm** [comment]
Hi, looks like [#5926](https://github.com/excalidraw/excalidraw/issues/5926) has been merged. Can I work on this one?

**Comment by dwelle** [comment]
Hi @jamiecjm, sure thing!

I'll send you the design spec tomorrow.

Implementation-wise, let's use the `@radix-ui/react-dropdown-menu` package we're using in https://github.com/excalidraw/excalidraw/pull/6483 (you don't need to wait for the PR to be merged and just install it directly).

**Comment by jamiecjm** [comment]
Thanks! What's the reason of not using [@radix-ui/react-context-menu](https://www.radix-ui.com/docs/primitives/components/context-menu)? Seems to be more straightforward implementation wise and the size is around [26.55 kB](https://bundlephobia.com/package/@radix-ui/react-context-menu@2.1.4)

**Comment by ad1992** [comment]
@jamiecjm the size seems too large for context menu, implementation wise I am hoping it should be similar using the menu as well. 
However if you want you can check how much bundle size is getting increased with `context-menu` by running `yarn size` in `src/packages/excalidraw` depending on our usage and based on that we can decide.

**Comment by jamiecjm** [comment]
Tested this using [#6483](https://github.com/excalidraw/excalidraw/pull/6483)

With `react-dropdown-menu`:

![Image](https://github.com/excalidraw/excalidraw/assets/26139842/69dbce86-b090-4767-a293-ffcf5f5eabc4)

With `react-context-menu`:

![Image](https://github.com/excalidraw/excalidraw/assets/26139842/4f8a94c8-605b-4c4a-bf0c-72d1c8486e0c)


I can use `react-dropdown-menu`, I'm just curious.

**Comment by ad1992** [comment]
so almost ~17kB increased, lets go ahead dropdown then  @jamiecjm :)

Effort wise I am hoping it would be similar ?

**Comment by jamiecjm** [comment]
Yeah it should be similar

**Comment by jamiecjm** [comment]
How does it look in dark mode? Does using the variables/design token adjust to dark/light mode automatically?

**Comment by ad1992** [comment]
> How does it look in dark mode? Does using the variables/design token adjust to dark/light mode automatically?

@jamiecjm yes the css variables will take care of dark mode, there are equivalent values in dark mode 
https://github.com/excalidraw/excalidraw/blob/master/src/css/theme.scss#L138

**Comment by dwelle** [comment]
@jamiecjm I've added the dark mode.

And by mistake the figma design I posted above was using a color scheme we will deploy later and does not reflect the current one in production. I've updated the file: https://www.figma.com/file/EvavjRtXUmHjrcNIBTAMxI/Public-components?type=design&node-id=32%3A4746&mode=design&t=yqNXPLjBOd49aMhh-1

**Comment by jamiecjm** [comment]
What should the hover state of Delete look like? The current behaviour has a red background on hover


![Image](https://github.com/excalidraw/excalidraw/assets/26139842/ba4b78bb-52d0-4890-a8d4-58f319ae1f2b)

**Comment by ad1992** [comment]
> What should the hover state of Delete look like? The current behaviour has a red background on hover
> 
> ![Image](https://user-images.githubusercontent.com/26139842/258259246-5656c098-9642-4e2d-8855-c45c16d66624.png)

lets keep it same as prod, we can tweak in PR is needed

**Comment by jamiecjm** [comment]
Hi should I use the chevron right icon from radix-ui? Or is there one in the package already?


![Image](https://github.com/excalidraw/excalidraw/assets/26139842/0cbc4b46-00da-4a87-a63b-af66d5f3f89c)

**Comment by aakashtyg** [comment]
@ad1992 is someone still working on this? Can I pick this up?

**Comment by EswaraSatya** [comment]
Hi ,

If no one started i want to take this up

**Comment by RafaelZasas** [comment]
Resurfacing this

**Comment by EswaraSatya** [comment]
> Resurfacing this

Can i work on this please let me know so that i can start work.

**Comment by MilosNj** [comment]
Hi @ad1992, is @jamiecjm still working on this? If not, I can pick up.

**Comment by Y1a2s3h4** [comment]
Hi @ad1992 @dwelle this is fixed I guess we should close this.

currently going through list of items which I can pick. Feel free to assign `good first issue`
